erts/etc/common/escript.c

```diff
@@ -106,35 +106,36 @@ get_env(char *key)
 {
 #ifdef __WIN32__
     DWORD size = 32;
-    char  *value=NULL;
-    wchar_t *wcvalue = NULL;
     wchar_t wckey[256];
-    int len; 
+    int len;
 
     MultiByteToWideChar(CP_UTF8, 0, key, -1, wckey, 256);
     
-    while (1) {
-	DWORD nsz;
-	if (wcvalue)
-	    efree(wcvalue);
-	wcvalue = (wchar_t *) emalloc(size*sizeof(wchar_t));
-	SetLastError(0);
-	nsz = GetEnvironmentVariableW(wckey, wcvalue, size);
-	if (nsz == 0 && GetLastError() == ERROR_ENVVAR_NOT_FOUND) {
-	    efree(wcvalue);
-	    return NULL;
-	}
-	if (nsz <= size) {
-	    len = WideCharToMultiByte(CP_UTF8, 0, wcvalue, -1, NULL, 0, NULL, NULL);
-	    value = emalloc(len*sizeof(char));
-	    WideCharToMultiByte(CP_UTF8, 0, wcvalue, -1, value, len, NULL, NULL);
-	    efree(wcvalue);
-	    return value;
-	}
-	size = nsz;
+    while (TRUE) {
+        DWORD nsz;
+        char *value = NULL;
+        wchar_t *wcvalue = (wchar_t *) emalloc(size*sizeof(wchar_t));
+        SetLastError(0);
+        nsz = GetEnvironmentVariableW(wckey, wcvalue, size);
+        if (nsz == 0 && GetLastError() == ERROR_ENVVAR_NOT_FOUND) {
+            efree(wcvalue);
+            return NULL;
+        }
+        if (nsz <= size) {
+            len = WideCharToMultiByte(CP_UTF8, 0, wcvalue, -1, NULL, 0, NULL, NULL);
+            value = emalloc(len*sizeof(char));
+            WideCharToMultiByte(CP_UTF8, 0, wcvalue, -1, value, len, NULL, NULL);
+            efree(wcvalue);
+            return value;
+        }
+
+        if (wcvalue)
+            efree(wcvalue);
+
+        size = nsz;
     }
 #else
-    return getenv(key);
+    return secure_getenv(key);
```

Comment:
The secure_getenv() function is intended for use in general-purpose libraries to avoid vulnerabilities that could occur if set-user-ID or set-group-ID programs accidentally trusted the environment.
- https://man7.org/linux/man-pages/man3/getenv.3.html

Reply:
`secure_getenv` is not available on all platforms that we support. So there needs to be a configure check to see if it is available.

Also, I'm unsure if it is a good idea to only change this single instance of `getenv`. If we do want to change this (which I'm unsure about as I don't know a lot about Linux security model), then we should change it everywhere at once so that we get consistent behaviour.